Update template/ .yaml files

milljm opened the pull request:
Bring template yaml files up-to-date with the rest of the conda/ packages.

Closes #21609

Commit:
Update template/ .yaml files

Bring template yaml files up-to-date with the rest of the conda/ packages.

Closes #21609

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1112926/) on 8934ddf wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
This will cause all the packages to be built. But I am not sure we should bump a build version or not. The packages built will just replace whats already available on /public with no 'real' change to those packages.

What do you guys think? Should we create a new set of packages or just leave it alone and have them replaced?

cticenhour commented:
Nothing fundamental about the packages have changed in terms of functionality or configuration, so I am happy to not bump the build numbers.

loganharbour commented:
Also fine with not upping the build numbers; but we will need to rebuild the `moose-*` package stack on the build boxes as their versioning is based on a SHA of this file, and a comment _is_ a change.

Commit:
pin Python version

Due to hit parser, we must pin runtime python to the same version of
python used during build.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1113127/) on ba2792b wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Be strict

Make runtime dependencies the same as what was used during buildtime.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1113443/) on eb049ec wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1113443/) on eb049ec wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops
